packages/common/src/associations/initEntityAssociationGroup.ts

```diff
@@ -0,0 +1,43 @@
+import { IArcGISContext } from "../ArcGISContext";
+import { IHubGroup, getTypeFromEntity } from "../core";
+import { HubEntity } from "../core/types/HubEntity";
+import { updateHubEntity } from "../core/updateHubEntity";
+import { setProp } from "../objects";
+
+/**
+ * Utility to initialize an entity association group that has just been created.
+ * This adds the entity-specific association keyword to the association group,
+ * and it adds the association definition to the original entity.
+ * @param entity
+ * @param group
+ * @param context
+ * @returns
+ */
+export async function initEntityAssociationGroup(
+  entity: HubEntity,
+  group: IHubGroup,
+  context: IArcGISContext
+): Promise<HubEntity> {
+  const type = getTypeFromEntity(entity);
+
+  // 1. Add the association keyword to the group entity
+  group.typeKeywords = [...group.typeKeywords, `${type}|${entity.id}`];
```

Comment:
If you add a guard to check if the keyword exists, then this could also work for using an existing group for the association. (or maybe you guard that in the UI layer where the Group selection query precludes groups with this keyword. Consider renaming the function as well b/c it's not "init"-ing the group - it's adding the association definition, and part of that is the keyword... iirc :)